src/components/Swap/Settings/index.tsx

```diff
@@ -1,72 +1,48 @@
-import { Trans } from '@lingui/macro'
+import { useCloseOnEscape } from 'hooks/useCloseOnEscape'
 import { Settings as SettingsIcon } from 'icons'
-import { useAtomValue, useResetAtom } from 'jotai/utils'
-import { useCallback, useState } from 'react'
-import { swapEventHandlersAtom } from 'state/swap'
-import { settingsAtom } from 'state/swap/settings'
+import { useState } from 'react'
 import styled from 'styled-components/macro'
-import { ThemedText } from 'theme'
 
-import { IconButton, TextButton } from '../../Button'
+import { IconButton } from '../../Button'
 import Column from '../../Column'
-import Dialog, { Header } from '../../Dialog'
-import { BoundaryProvider } from '../../Popover'
+import Popover, { PopoverBoundaryProvider } from '../../Popover'
 import MaxSlippageSelect from './MaxSlippageSelect'
 import TransactionTtlInput from './TransactionTtlInput'
 
-export function SettingsDialog() {
+export function SettingsPopover() {
   const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
-  const { onSettingsReset } = useAtomValue(swapEventHandlersAtom)
-  const resetSettingsBase = useResetAtom(settingsAtom)
-  const resetSettings = useCallback(() => {
-    onSettingsReset?.()
-    resetSettingsBase()
-  }, [onSettingsReset, resetSettingsBase])
+
+  // TODO: add back reset settings functionality
   return (
-    <>
-      <Header title={<Trans>Settings</Trans>} ruled>
-        <TextButton onClick={resetSettings}>
-          <ThemedText.ButtonSmall>
-            <Trans>Reset</Trans>
-          </ThemedText.ButtonSmall>
-        </TextButton>
-      </Header>
-      <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded>
-        <BoundaryProvider value={boundary}>
-          <MaxSlippageSelect />
-          <TransactionTtlInput />
-        </BoundaryProvider>
-      </Column>
-    </>
+    <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded>
+      <PopoverBoundaryProvider value={boundary}>
+        <MaxSlippageSelect />
+        <TransactionTtlInput />
+      </PopoverBoundaryProvider>
+    </Column>
   )
 }
 
-const SettingsButton = styled(IconButton)<{ hover: boolean }>`
-  ${SettingsIcon} {
-    transform: ${({ hover }) => hover && 'rotate(45deg)'};
-    transition: ${({ hover }) => hover && 'transform 0.25s'};
-    will-change: transform;
+const SettingsButton = styled(IconButton)`
+  ${SettingsIcon}:hover {
+    transform: rotate(45deg);
+    transition: transform 0.25s;
```

Comment:
This is a regression, as it will not rotate back onmouseleave. What if you move transition outside of the hover pseudo-selector?

Reply:
it doesn't currently rotate back on onmouseleave i believe? at least not when i try it on the token details page: https://app.uniswap.org/#/tokens/ethereum/0xf1b99e3e573a1a9c5e6b2ce818b617f0e664e86b

i can add it though